ifsbench/data/namelisthandler.py

```diff
@@ -0,0 +1,167 @@
+        ----------
+        namelist: f90nml.Namelist
+            The namelist to which the changes are applied.
+        """
+
+        if self._keys[0] not in namelist:
+            if self._mode == self.NamelistOperation.DELETE:
+                return
+
+            namelist[self._keys[0]] = {}
+
+        namelist = namelist[self._keys[0]]
+        key = self._keys[-1]
+
+        if self._mode == self.NamelistOperation.SET:
+            debug(f"Set namelist entry {str(self._keys)} = {str(self._value)}.")
+            namelist[key] = self._value
+        elif self._mode == self.NamelistOperation.APPEND:
+            if key not in namelist:
+                namelist[key] = []
+
+            if not hasattr(namelist[key], 'append'):
+                raise ValueError("Values can only be appended to arrays!")
+
+            # f90nml doesn't seem to do any kind of checking, so we could
+            # create arrays in the namelist where the entries have different
+            # types.
+            # This will most likely cause issues, so we verify here, that
+            # the array entries have the same type.
+            if len(namelist[key]) > 0:
+                type_list = type(namelist[key][0])
+                type_value = type(self._value)
+
+                if type_list != type_value:
+                    raise ValueError("The given value must have the same type as existing array entries!")
+
+            debug(f"Append {str(self._value)} to namelist entry {str(self._keys)}.")
+
+            namelist[key].append(self._value)
+
+        elif self._mode == self.NamelistOperation.DELETE:
+            if key in namelist:
+                debug(f"Delete namelist entry {str(self._keys)}.")
+                del namelist[key]
+
+class NamelistHandler(DataHandler):
+    """
+    DataHandler specialisation that can modify Fortran namelists.
+    """
+
+    def __init__(self, input_path, output_path, overrides):
+        """
+        Initialise the handler.
+
+        Parameters
+        ----------
+        input_path: str or `pathlib.Path`
+            The path to the namelist that will be modified. If a relative path
+            is given, this will be relative to the `wdir` argument in `execute`.
+
+        output_path: str, `pathlib.Path` or `None`
+            The path to which the updated namelist will be written. If a relative path
+            is given, this will be relative to the `wdir` argument in `execute`.
+            If None is given, this will re extracted to `wdir`.
+
+        overrides: iterable of NamelistOverride
+            The NamelistOverrides that will be applied.
+        """
+
+        self._input_path = pathlib.Path(input_path)
+        self._output_path = pathlib.Path(output_path)
+
+        self._overrides = list(overrides)
+        for override in self._overrides:
+            if not isinstance(override, NamelistOverride):
+                raise ValueError("Namelist overrides must be NamelistOverride objects!")
```

Comment:
You could use a list comprehension here:
```suggestion
        if not all(isinstance(override, NamelistOverride) for override in self._overrides):
            raise ValueError("Namelist overrides must be NamelistOverride objects!")
```